lt/client.go

```diff
@@ -0,0 +1,639 @@
+		}
+		defer file.Close()
+
+		ivf, header, ivfErr := ivfreader.NewWith(file)
+		if ivfErr != nil {
+			log.Fatalf(ivfErr.Error())
+		}
+
+		<-time.After(2 * time.Second)
+		// Wait for connection established
+		<-connectedCh
+
+		info := map[string]string{
+			"screenStreamID": track.StreamID(),
+		}
+		data, err := json.Marshal(&info)
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+
+		if err := ws.SendMessage("custom_com.mattermost.calls_screen_on", map[string]interface{}{
+			"data": string(data),
+		}); err != nil {
+			log.Fatalf(err.Error())
+		}
+		defer func() {
+			if err := ws.SendMessage("custom_com.mattermost.calls_screen_off", nil); err != nil {
+				log.Fatalf(err.Error())
+			}
+		}()
+
+		// Send our video file frame at a time. Pace our sending so we send it at the same speed it should be played back as.
+		// This isn't required since the video is timestamped, but we will such much higher loss if we send all at once.
+		//
+		// It is important to use a time.Ticker instead of time.Sleep because
+		// * avoids accumulating skew, just calling time.Sleep didn't compensate for the time spent parsing the data
+		// * works around latency issues with Sleep (see https://github.com/golang/go/issues/44343)
+		ticker := time.NewTicker(time.Millisecond * time.Duration((float32(header.TimebaseNumerator)/float32(header.TimebaseDenominator))*1000))
+		for ; true; <-ticker.C {
+			var frame []byte
+			var ivfErr error
+			frame, _, ivfErr = ivf.ParseNextFrame()
+			if ivfErr == io.EOF || (ivfErr != nil && ivfErr.Error() == "incomplete frame data") {
+				ivf.ResetReader(func(_ int64) io.Reader {
+					_, _ = file.Seek(0, 0)
+					ivf, header, ivfErr = ivfreader.NewWith(file)
+					if ivfErr != nil {
+						log.Fatalf(ivfErr.Error())
+					}
+					return file
+				})
+				frame, _, ivfErr = ivf.ParseNextFrame()
+			}
+			if ivfErr != nil {
+				log.Fatalf(ivfErr.Error())
+			}
+
+			if err := track.WriteSample(media.Sample{Data: frame, Duration: time.Second}); err != nil {
+				log.Printf("failed to write video sample: %s", err.Error())
+			}
+		}
+	}()
+}
+
+func transmitAudio(ws *websocket.Client, pc *webrtc.PeerConnection, connectedCh <-chan struct{}) {
+	track, err := webrtc.NewTrackLocalStaticSample(rtpAudioCodec, "audio", "voice"+model.NewId())
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	rtpSender, err := pc.AddTrack(track)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	go func() {
+		rtcpBuf := make([]byte, 1500)
+		for {
+			if _, _, rtcpErr := rtpSender.Read(rtcpBuf); rtcpErr != nil {
+				return
+			}
+		}
+	}()
+
+	go func() {
+		// Open a OGG file and start reading using our OGGReader
+		file, oggErr := os.Open("./lt/samples/audio.ogg")
+		if oggErr != nil {
+			log.Fatalf(oggErr.Error())
+		}
+		defer file.Close()
+
+		// Open on oggfile in non-checksum mode.
+		ogg, _, oggErr := oggreader.NewWith(file)
+		if oggErr != nil {
+			log.Fatalf(oggErr.Error())
+		}
+
+		<-time.After(2 * time.Second)
```

Comment:
@cpoile Could you remind me why we have to sleep for a bit here? I've kept the semantic you introduced but forgot what we were trying to achieve (or avoid that is).